lib/spack/spack/environment.py

```diff
@@ -436,7 +477,7 @@ def return_separator_if_any(*args):
                         env.prepend_path(x, item)
             else:
                 # We just need to set the variable to the new value
-                env.set(x, env_after[x])
+                env.set(x, after)
```

Comment:
I am not sure if this could make a difference, but in your PR the variable `after` gets reassigned and is not exactly what you read in the environment *after* sourcing a file - to be clear it's split, deduped and rejoined. This is just to say: let's keep an eye on this part if we use this function in packages other than OpenFOAM. Intel* could be a good candidate for further study in later PRs.

Reply:
Yes this is actually what I wanted, since I ended up with really horrible results (see commit message on 66be6b556519d227718a03585586dd4695ec3852). So at the very least I'll need to make this another flag. I'll get on these changes now.

Reply:
Keep it like that for now. If problem arises down the road we'll see how to deal with them.

Reply:
OK, but between reading the messages, it's already done and pushed.